dsl/convertcty.go

```diff
@@ -0,0 +1,79 @@
+package dsl
+
+import (
+	"fmt"
+
+	"github.com/zclconf/go-cty/cty"
+)
+
+// convertCtyValueToInterface converts a cty.Value to an interface{}.
+//
+// Calls itself recursively to convert nested values.
+// Does not cover all possible cty types, such as unknown, capsule, empty object,
+// and empty tuple.
+func convertCtyValueToInterface(val cty.Value) (interface{}, error) {
+	if val.IsNull() {
+		return nil, nil
+	}
+
+	switch {
+
+	case val.Type().Equals(cty.String):
```

Comment:
val.Type() can be done once at the top of the switch statement and then have `case: cty.String` etc instead

Reply:
https://github.com/zclconf/go-cty/issues/158#issuecomment-1694704006

Reply:
I think some of the types can't be done that way. I can get type once however.

Here's their code from [`go-cty/cty](https://github.com/zclconf/go-cty/tree/v1.13.2/cty)
/type.go`](https://github.com/zclconf/go-cty/blob/v1.13.2/cty/type.go#L121C1-L154):

```go
func (t Type) WithoutOptionalAttributesDeep() Type {
	switch {
	case t == DynamicPseudoType, t.IsPrimitiveType(), t.IsCapsuleType():
		return t
	case t.IsMapType():
		return Map(t.ElementType().WithoutOptionalAttributesDeep())
	case t.IsListType():
		return List(t.ElementType().WithoutOptionalAttributesDeep())
	case t.IsSetType():
		return Set(t.ElementType().WithoutOptionalAttributesDeep())
	case t.IsTupleType():
		originalElemTypes := t.TupleElementTypes()
		elemTypes := make([]Type, len(originalElemTypes))
		for i, et := range originalElemTypes {
			elemTypes[i] = et.WithoutOptionalAttributesDeep()
		}
		return Tuple(elemTypes)
	case t.IsObjectType():
```

Reply:
Fixed to get type only once.